assets/js/event-filtering.js

```diff
@@ -0,0 +1,67 @@
+$(function() {
+    var eventsElements = $(".event-tags span");
+    var eventFilterParent = $("#eventFilter");
+    var tags = [];
+
+    for (var i = 0; i < eventsElements.length; i++) {
+        var elem = eventsElements[i];
+        var text = $(elem).text();
+        if (tags.indexOf(text) === -1) {
+            tags.push(text);
+            eventFilterParent.append(
+                '<input type="checkbox" class="mx-2" value="' + text + '">' + text + '</p>'
```

Comment:
What if the value of text is somehow `<script>alert("lol, y'all been hacked")</script>`? Is there a jQuery method for sanitizing input in situations like this? (Just to err on the side of belt+suspenders.)

Reply:
I adjusted the code to generate the elements via the document API and user `innerText` to set the value of the text. So that should cover some malicious situation where the `tag` value is a script like you mentioned above. Everything will just get rendered as a string .